### ISSUE TITLE
Broken code coverage (using meteor-coverage) since meteor 1.6

### ISSUE THREAD

**Issue by SimonSimCity** [meteor/meteor#9281]
Even though I'm talking about a 3rd party plugin - let me explain how this bug is introduced by Meteor.

The plugin `meteor-coverage` uses `istanbul-api` to gather code coverage information. In order to listen for the coverage, it uses [hooks on the internal `vm`](https://github.com/istanbuljs/istanbuljs/blob/master/packages/istanbul-lib-hook/lib/hook.js) module to get this information; one of these hooks was `vm.runInThisContext()` which was [(until this change got merged)](https://github.com/meteor/meteor/commit/f244a5b0c63a7e2b961894db2260b6f9e0de08f6) used to load the files in meteor. The new method (using `vm.Script.runInThisContext()`) is not supported by the time writing.

1. This means, that `meteor-coverage` currently has no valid entry point to register itself.

After I figured that out, I wondered why I still don't get any data about the coverage. Here the devil is in the detail 😉 . The initialization of `meteor-coverage` (which calls the hook to wrap the`vm.runInThisContext()` method into code-coverage) got postponed by [this change](https://github.com/meteor/meteor/commit/8550412bb0eaa3b80ae8725f3bed128d629b9d61#diff-e85840f89d0889ea2a4034009dbccc05R170) while introducing the node-debugger.

2. This means, the hook for code-coverage is initialized after all modules are loaded - which leaves no modules for code-coverage.

I hope you understand my concerns here and I can count on your help @benjamn @abernix and @serut to get it working again 😢 

You can simply check it by starting one of your projects, add `meteor-coverage` to it and open `http://localhost:3000/coverage` in your browser, or add some logs into the `meteor-coverage` package - like I did in https://github.com/serut/meteor-coverage/issues/54.

**Comment by hwillson** [comment]
@SimonSimCity Based on https://github.com/serut/meteor-coverage/issues/54#issuecomment-343734991, it sounds like this can be resolved without making any changes to Meteor. I'll close this for now, but let me know if I'm missing anything and this should be kept open. Thanks!

**Comment by SimonSimCity** [comment]
@hwillson please reopen this issue, as https://github.com/serut/meteor-coverage/issues/54 doesn't solve it and it still needs some adjustment on Meteor - as far as I understand the issue.

**Comment by hwillson** [comment]
Will do @SimonSimCity - I'll keep an eye on https://github.com/serut/meteor-coverage/issues/54 as well.

**Comment by hwillson** [comment]
A quick update here for reference - the `meteor-coverage` project is waiting on https://github.com/istanbuljs/istanbuljs/pull/99 to be accepted by `istanbuljs`. After that's done, the work to make `meteor-coverage` Meteor  1.6 compliant will be completed.
